server/container/guice/protocols/smtp/src/main/java/org/apache/james/modules/protocols/SmtpGuiceProbe.java

```diff
@@ -36,10 +38,19 @@ private SmtpGuiceProbe(SMTPServerFactory smtpServerFactory) {
     }
 
     public int getSmtpPort() {
+        return findFirstPort(server -> true);
+    }
+
+    public int getSmtpPort(Predicate<? super AbstractConfigurableAsyncServer> filter) {
```

Comment:
Ouch... As a caller do I really need to know the structure of a SMTP server?

Can't we not push this knowledge to the client?

I i,agine what you truely want is a `getSmtpsPort()` method...

Reply:
I updated the fixup, please check https://github.com/linagora/james-project/pull/1640/commits/88a15a6f6d875144930fef3489490d40a98807a0